lib/phoenix/channel/transport.ex

```diff
@@ -95,28 +95,42 @@ defmodule Phoenix.Channel.Transport do
                   endpoint: endpoint,
                   pubsub_server: endpoint.__pubsub_server__(),
                   topic: msg.topic,
-                  ref: msg.ref,
                   channel: channel,
                   transport: transport}
 
-        Phoenix.Channel.Server.start_link(socket, msg.payload)
+        case Phoenix.Channel.Server.join(socket, msg.payload) do
```

Comment:
We will need to change this for long polling so we return the replies on the same request.